include/aos/iam/certmodule.hpp

```diff
@@ -0,0 +1,378 @@
+     */
+    bool mSkipValidation;
+};
+
+/**
+ * General certificate information.
+ */
+struct CertInfo {
+    /**
+     * Certificate issuer name.
+     */
+    StaticString<crypto::cCertIssuerLen> mIssuer;
+    /**
+     * Certificate serial number.
+     */
+    StaticString<crypto::cCertSerialNumberLen> mSerial;
+    /**
+     * Certificate url.
+     */
+    StaticString<cURLLen> mCertURL;
+    /**
+     * Certificate's private key url.
+     */
+    StaticString<cURLLen> mKeyURL;
+    /**
+     * Certificate expiration time.
+     */
+    time::Time mNotAfter;
+    /**
+     * Checks whether certificate info is equal the the current one.
+     *
+     * @param certInfo info to compare.
+     * @return bool.
+     */
+    bool operator==(const CertInfo& certInfo) const
+    {
+        return certInfo.mCertURL == mCertURL && certInfo.mIssuer == mIssuer && certInfo.mKeyURL == mKeyURL
+            && certInfo.mNotAfter == mNotAfter && certInfo.mSerial == mSerial;
+    }
+    /**
+     * Checks whether certificate info is equal the the current one.
+     *
+     * @param certInfo info to compare.
+     * @return bool.
+     */
+    bool operator!=(const CertInfo& certInfo) const { return !operator==(certInfo); }
+
+    /**
+     * Outputs object to log.
+     *
+     * @param log log to output.
+     * @param certInfo object instance.
+     * @return Log&.
+     */
+    friend Log& operator<<(Log& log, const CertInfo& certInfo)
+    {
+        log << "[certURL: " << certInfo.mCertURL << ", keyURL: " << certInfo.mKeyURL
+            << ", notAfter: " << certInfo.mNotAfter << "]";
+        return log;
+    }
+};
+
+/**
+ * Platform dependent secure certificate storage.
+ */
+class HSMItf {
+public:
+    /**
+     * Owns the module.
+     *
+     * @param password certficate password.
+     * @return Error.
+     */
+    virtual Error SetOwner(const String& password) = 0;
+
+    /**
+     * Removes all module certificates.
+     *
+     * @return Error.
+     */
+    virtual Error Clear() = 0;
+
+    /**
+     * Generates private key.
+     *
+     * @param password owner password.
+     * @return RetWithError<crypto::PrivateKey*>.
+     */
+    virtual RetWithError<crypto::PrivateKey*> CreateKey(const String& password, KeyGenAlgorithm algorithm) = 0;
+
+    /**
+     * Applies certificate chain to a module.
+     *
+     * @param certChain certificate chain.
+     * @param[out] topCertInfo info about a top certificate in a chain.
+     * @param[out] password owner password.
+     * @return Error.
+     */
+    virtual Error ApplyCert(const Array<crypto::x509::Certificate>& certChain, CertInfo& topCertInfo, String& password)
+        = 0;
+
+    /**
+     * Removes certificate chain using top level certificate URL and password.
+     *
+     * @param certURL top level certificate URL.
+     * @param password owner password.
+     * @return Error.
+     */
+    virtual Error RemoveCertChain(const String& certURL, const String& password) = 0;
```

Comment:
Rename `RemoveCertChain` to `RemoveCert` it should match `ApplyCert`.

Reply:
Done